### PR TITLE
handle null ComputerName 

### LINKED ISSUE #6632: [Bug] Microsoft.SqlServer.Management.Smo.Server object may return null computername
### Environmental information

```powershell
LCID             Name             DisplayName
----             ----             -----------
2052             zh-CN            中文(简体，中国)

```

#### SQL Server:

```sql
Microsoft SQL Server 2016 (SP2) (KB4052908) - 13.0.5026.0 (X64)   Mar 18 2018 09:11:49   Copyright (c) Microsoft Corporation  Developer Edition (64-bit) on Windows Server 2019 Standard 10.0 <X64> (Build 17763: ) (Hypervisor) 
```

```sql
简体中文
```

#### Host used

- [X] powershell.exe
- [ ] ISE
- [ ] VS Code
- [ ] Other (please specify)

If anything other than powershell.exe was used, please confirm that you can duplicate the issue with powershell.exe

- [x] Still buggy in powershell.exe

#### Errors Received

```
警告: [10:23:35][Get-DecryptedObject] Can't access registry keys on mydb. Do you have administrative access to the
Windows registry on [mydb] Otherwise, we're out of ideas.
```

#### Steps to Reproduce

```powershell
$dbatools=Import-Module dbatools -PassThru
& $dbatools{ Get-DecryptedObject -SqlInstance mydb -Type LinkedServer -Verbose}
```

#### Expected Behavior

i can get decrypted linkedserver list with password,but it always report an error like the above,so i add some debug in allcommands.ps1, and i found Microsoft.SqlServer.Management.Smo.Server object may return null computername. so the internal function like resolve-netbiosname、resolve-sqlipaddress should handle such situation.

#### Actual Behavior

```
详细信息: [10:26:17][Get-DecryptedObject] Querying service master key
详细信息: [mydb]
详细信息: myDB
详细信息: Microsoft.SqlServer.Management.Smo.Server
详细信息:
ComputerName Name Product              Version   HostPlatform IsAzure IsClustered ConnectedAs
------------ ---- -------              -------   ------------ ------- ----------- -----------
             mydb Microsoft SQL Server 13.0.4422 Windows              False


Write-Verbose : 无法将参数绑定到参数“Message”，因为该参数是空值。
所在位置 C:\Windows\system32\WindowsPowerShell\v1.0\Modules\dbatools\allcommands.ps1:80651 字符: 19
+     write-verbose $server.ComputerName
+                   ~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidData: (:) [Write-Verbose]，ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationErrorNullNotAllowed,Microsoft.PowerShell.Commands.WriteVerbos
   eCommand

Write-Verbose : 无法将参数绑定到参数“Message”，因为该参数是空值。
所在位置 C:\Windows\system32\WindowsPowerShell\v1.0\Modules\dbatools\allcommands.ps1:76456 字符: 19
+     write-verbose $sourceNetBios
+                   ~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidData: (:) [Write-Verbose]，ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationErrorNullNotAllowed,Microsoft.PowerShell.Commands.WriteVerbos
   eCommand

详细信息: [10:26:17][Get-DecryptedObject] Get entropy from the registry - hopefully finds the right SQL server instance
Get-DecryptedObject : 无法对参数“ComputerName”执行参数验证。参数为 Null 或空。请提供一个不为 Null 或空的参数，然后重
试该命令。
所在位置 行:1 字符: 14
+ & $dbatools{ Get-DecryptedObject -SqlInstance mydb -Type LinkedServer -Verbose}
+              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,Get-DecryptedObject

警告: [10:26:17][Get-DecryptedObject] Can't access registry keys on mydb. Do you have administrative access to the
Windows registry on [mydb] Otherwise, we're out of ideas.
```

### PR DESCRIPTION
**Pull request by yoke88**
## Type of Change

 - [x] Bug fix (non-breaking change, fixes #6632 )
 - [ ] New feature (non-breaking change, adds functionality, fixes # )
 - [ ] Breaking change (effects multiple commands or functionality, fixes # )
 - [ ] Ran manual Pester test and has passed (`.\tests\manual.pester.ps1)
 - [ ] Adding code coverage to existing functionality
 - [ ] Pester test is included
 - [ ] If new file reference added for test, has is been added to github.com/sqlcollaborative/appveyor-lab ?
 - [ ] Nunit test is included
 - [ ] Documentation
 - [ ] Build system
 

### Purpose
`[Microsoft.SqlServer.Management.Smo.Server` may return null computername ,so Resolve-NetBiosName、Resolve-SqlIpAddress will return null too, and cause other function fails.

``` powershell

PS C:\> [Microsoft.SqlServer.Management.Smo.Server]"mydb"|fl *name*


ClusterName                 :
ComputerNamePhysicalNetBIOS : MYDB
FilestreamShareName         : MSSQLSERVER
InstanceName                :
NamedPipesEnabled           : False
NetName                     : MYDB
ServiceName                 : MSSQLSERVER
SqlCharSetName              : cp936
SqlSortOrderName            : bin_ascii_8
Name                        : mydb
DomainName                  : SMO
DomainInstanceName          : MYDB

```

### Approach
modify Resolve-NetBiosName and Resolve-SqlIpAddress to handle null computername .

### Screenshots
image

### CODE REVIEW

File: internal/functions/Get-DecryptedObject.ps1 [hunk 1]
```diff
@@ -22,7 +22,7 @@ function Get-DecryptedObject {
     # Query Service Master Key from the database - remove padding from the key
     # key_id 102 eq service master key, thumbprint 3 means encrypted with machinekey
     Write-Message -Level Verbose -Message "Querying service master key"
-    $sql = "SELECT substring(crypt_property,9,len(crypt_property)-8) as smk FROM sys.key_encryptions WHERE key_id=102 and (thumbprint=0x03 or thumbprint=0x0300000001)"
+    $sql = "SELECT substring(crypt_property,9,datalength(crypt_property)-8) as smk FROM sys.key_encryptions WHERE key_id=102 and (thumbprint=0x03 or thumbprint=0x0300000001)"
```
Comment (wsmelton) [inline internal/functions/Get-DecryptedObject.ps1:25]: What purpose does this change serve?

Comment (wsmelton) [reply internal/functions/Get-DecryptedObject.ps1:25]: LEN returns the number of characters in the string, DATALENGTH returns the number of bytes.

Comment (yoke88) [reply internal/functions/Get-DecryptedObject.ps1:25]: i just see it here and did a test on my env and it works:
https://github.com/bugal/Powershell-Modules/commit/606153b890f831e94890ef85793451b28ead0433

there are other fix to workground for SQL cluster too ,you may need this https://github.com/NetSPI/Powershell-Modules/pull/9

**Comment by wsmelton** [comment]
I'm closing this PR as this change and the error provided are not reproducible by our test or manual testing I've done in my lab.

Discussion can continue in your issue.

**Pull request closed without merging**